### PR TITLE
Bring the API to match Elastica 3.0.1

### PR DESCRIPTION
**Pull request by emgiezet**
Hi!,

I made most of the changes. But still one test don't pass. Can anybody help me with the:  MappingToElasticaTest:testResetIndexAddsMappings

The Bundle work with 14 GB Index (production dump tested) But still test fails.

Regards!
Max

### CODE REVIEW

**Comment by kamdz** [comment]
:+1:

**Comment by kamillo1888** [comment]
:+1:

**Comment by sebastiancichos** [comment]
+1 we need this too...

**Comment by emgiezet** [comment]
Any luck with this?

**Comment by dunglas** [comment]
I confirm that it works for me too.

File: Tests/Functional/MappingToElasticaTest.php [hunk 1]
```diff
@@ -21,11 +21,13 @@ class MappingToElasticaTest extends WebTestCase
     public function testResetIndexAddsMappings()
     {
         $client = $this->createClient(array('test_case' => 'Basic'));
-        $resetter = $this->getResetter($client);
-        $resetter->resetIndex('index');
 
         $type = $this->getType($client);
         $mapping = $type->getMapping();
+        $resetter = $this->getResetter($client);
```
Comment (pascaldevink) [inline Tests/Functional/MappingToElasticaTest.php]: It seems that by juggling the location of the resetter, ```$this->getType($client)``` is using an index that hasn't been created yet. Resetting this to the original position fixes the current test error.

Comment (pascaldevink) [inline Tests/Functional/MappingToElasticaTest.php]: However, a lot of other things have been deprecated in ES2.x, so other errors will come up after that.

Comment (pascaldevink) [inline Tests/Functional/MappingToElasticaTest.php]: I've done a PR against your PR (PR-inception!) to address these issues: https://github.com/emgiezet/FOSElasticaBundle/pull/1

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -417,8 +417,8 @@ protected function getParentNode()
         $node
             ->children()
                 ->scalarNode('type')->end()
-                ->scalarNode('property')->defaultValue(null)->end()
-                ->scalarNode('identifier')->defaultValue('id')->end()
+//                ->scalarNode('property')->defaultValue(null)->end()
+//                ->scalarNode('identifier')->defaultValue('id')->end()
```
Comment (XWB) [inline DependencyInjection/Configuration.php]: This should be removed.

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -417,8 +417,8 @@ protected function getParentNode()
         $node
             ->children()
                 ->scalarNode('type')->end()
-                ->scalarNode('property')->defaultValue(null)->end()
-                ->scalarNode('identifier')->defaultValue('id')->end()
+//                ->scalarNode('property')->defaultValue(null)->end()
```
Comment (XWB) [inline DependencyInjection/Configuration.php]: This should be removed.

File: Paginator/RawPartialResults.php [hunk 4]
```diff
@@ -40,13 +40,10 @@ public function getTotalHits()
 
     /**
      * {@inheritDoc}
+     * @deprecated remvem from rufflin api 3.0.1
```
Comment (XWB) [inline Paginator/RawPartialResults.php]: Typo?

File: composer.json [hunk 5]
```diff
@@ -12,12 +12,12 @@
         { "name": "Jeremy Mikola", "email": "jmikola@gmail.com" }
     ],
     "require": {
-        "php": ">=5.3.2",
+        "php": ">=5.4.0",
```
Comment (XWB) [inline composer.json]: Why would we bump PHP to 5.4?

File: composer.json [hunk 6]
```diff
@@ -12,12 +12,12 @@
...
         "symfony/framework-bundle": "~2.3|~3.0",
         "symfony/console": "~2.1|~3.0",
         "symfony/form": "~2.1|~3.0",
         "symfony/property-access": "~2.3|~3.0",
-        "ruflin/elastica": "~2.1",
+        "ruflin/elastica": "3.0.*",
```
Comment (XWB) [inline composer.json]: Not sure if we want to drop Elastic 1.7 support, ping @merk

File: .travis.yml [hunk 7]
```diff
@@ -5,7 +5,6 @@ cache:
         - $HOME/.composer/cache
 
 php:
-  - 5.3
```
Comment (XWB) [inline .travis.yml]: What exactly goes wrong on PHP 5.3?

Comment (emgiezet) [inline .travis.yml]: The Es 3.0.1 have 5.4 in dependencies

Comment (XWB) [inline .travis.yml]: I see.

Just noticed PHP 5.3 is EOL http://php.net/supported-versions.php, I suppose now is a good time to drop 5.3 support.

**Comment by emgiezet** [comment]
i've rebased pr to newest master.

**Comment by merk** [comment]
Not willing to drop Elastica 2.x support or PHP 5.3 support until we bump major versions of FOSEB.

**Comment by emgiezet** [comment]
@merk this is sad :(

**Comment by dunglas** [comment]
@merk Support for Elastic 3 can maybe be added to another branch?

**Comment by merk** [comment]
The version required for using this bundle should not be locked down by what version of Elastica we want to support. The fact that Elastica has hard dependencies on specific ES versions means that we're locking out users who cannot upgrade from ES1.7.

This is not acceptable. I'm happy to merge any code that maintains BC. A separate branch is a no go, we do not have enough contributors or time allocated to management of this bundle to maintain 2 branches.

**Comment by emgiezet** [comment]
I'm using foselastica with a huge set of data. We implemented own populate command to gain multithreaded performance.
We skipped the listener transformer and put data to the ES server by async http request with guzzle and batch requests.

But still I can't see the logic after the decision to not follow version of your main and only dependant package.

I'm using elasticsearch since 0.6 beta. The changelog from 1.7 compared with current version is huge (ex. populating straight from mongo).

In this case I wish you guys luck. I must use pure rufflin/elastica instead,

And when you wondering why people don't contribute much. The answer is simple. 

> Because they don't want to waste their work.

Cheers!
Max

**Comment by pascaldevink** [comment]
@merk So that means that nobody can upgrade their ES cluster beyond 1.7.x if using this bundle? That would be a great way to loose almost everybody that is currently using it and not gaining any new users, as they would start out with the current ES version (2.2.x).

Wouldn't it be better to start working on a new major version with support for newer ES version? If people still want to use older versions of ES, and this bundle, that's fine, but let's try to cater more to the people that use the latest versions instead of trying to keep backwards compatibility for the people that refuse to upgrade (for whatever reason).

PHP 5.3 support was dropped over a year ago and ES 5 is already announced.

**Comment by merk** [comment]
So should we abandon users who don't have the capability to upgrade every time a new version of a product comes out? Symfony 2.3 is still supported and runs on PHP 5.3.

The truth is we use none of the new features of either Elastica or ElasticSearch and it should not be such a pain to support multiple versions. I've yet to see anything specific that cant be handled without breaking BC.

Theres a major problem: the bundle is doing too much in a way that just doesn't make it very useful for anything except a single case. Do you continue to develop this code base, which is sorely missing tests, or look at a new approach? I've not see many people comment when I've tried to raise the topic, nor offer to assist in making the situation better.

My contribution to Elastica to make it simpler exposes a big issue: since FOSEB doesnt actually use much in the way of Elastica's features, why dont we move to just using the official client? It already supports some of the things we need for the bundle.

https://github.com/ruflin/Elastica/pull/1017
https://github.com/FriendsOfSymfony/FOSElasticaBundle/issues/986

**Comment by kwojc** [comment]
@merk @XWB Last release of PHP 5.3 was in 2014, don't be silly, we are in 2016. If somebody can't bump up version of PHP, he should use older version of FOSEB IMO, next major version of FOSEB is needed urgently, people likes to use fresh vendors with support to newest versions of software.

**Comment by pawelkolanowski** [comment]
@kwojc +1

**Comment by merk** [comment]
You've missed the point here, dropping 5.3 isnt necessary for this PR. The actual point is I'm not dropping BC at this time. 

I'm actually tempted to deprecate the entire bundle since there is almost nothing in the way of help coupled with a lack of any real test coverage. It might be simpler to start over.

**Comment by Gladhon** [comment]
@merk 
Dropping 5.3 is necessary to update elasatic libary. If you cant update your own dependencies, you always stay on this buggy and feature-less version.

**Comment by mojzis** [comment]
@merk i think i understand most of your points, but still we suffer from not being able to use some stuff in ES 2 ... how about a compromise where this would go **temporarily** into a new branch that would also be a new major release and all the cleanup and refactoring would be done in yet another major release ?
Trying to support both versions of Elastica looks like something noone is willing to contribute now and since there is the working code from @emgiezet I think it would be a pity not to use it for the good of many projects dependent on FOSEB.
Thanks for considering the compromise :).

File: .travis.yml (hunk 7 above)
Comment (Soullivaneuh) [inline .travis.yml]: > Just noticed PHP 5.3 is EOL http://php.net/supported-versions.php, I suppose now is a good time to drop 5.3 support.

Yes, but it's have to be done a new major.

File: composer.json (hunk 6 above)
Comment (Soullivaneuh) [inline composer.json]: `3.0.*` -> `^3.0`.

> Not sure if we want to drop Elastic 1.7 support, ping @merk

Any chance to support both version of `ruflin/elastica` to make it BC?

**Comment by XWB** [comment]
Closing this in favor of https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1069

**Pull request closed without merging**